bpf/bpf_netdev.c

```diff
@@ -544,26 +568,34 @@ static __always_inline int do_netdev_encrypt_fib(struct __sk_buff *skb, int *enc
 	return ret;
 }
 
-static __always_inline int do_netdev_encrypt(struct __sk_buff *skb)
+static __always_inline int do_netdev_encrypt(struct __sk_buff *skb, __u16 proto)
 {
-	int encrypt_iface;
+	int encrypt_iface = 0;
 	int ret = 0;
 
 #ifdef ENCRYPT_NODE
 	encrypt_iface = ENCRYPT_IFACE;
 #endif
+#ifdef HAVE_FIB_LOOKUP
+	encrypt_iface = ENCRYPT_IFACE;
+#endif
 
 	ret = do_netdev_encrypt_pools(skb);
 	if (ret)
 		return send_drop_notify_error(skb, 0, ret, TC_ACT_SHOT, METRIC_INGRESS);
 
-	ret = do_netdev_encrypt_fib(skb, &encrypt_iface);
+	ret = do_netdev_encrypt_fib(skb, proto, &encrypt_iface);
 	if (ret)
 		return send_drop_notify_error(skb, 0, ret, TC_ACT_SHOT, METRIC_INGRESS);
 
 	bpf_clear_cb(skb);
-#ifdef ENCRYPT_NODE
+#if defined(ENCRYPT_NODE)
 	return redirect(encrypt_iface, 0);
+#elif defined(HAVE_FIB_LOOKUP)
+	ret = redirect(ENCRYPT_IFACE, 0);
+	if (ret)
+		bpf_printk("redirect return %d encrypt@ %d\n", ret, ENCRYPT_IFACE);
```

Comment:
More debugging here, is this intended in the final version?

Reply:
oops yeah it needed one more force push it seems I forgot an apply/push sequence its good (e.g. without any printks) on my local branch but got here with the previous revision.